src/backend/distributed/planner/recursive_planning.c

```diff
@@ -1305,6 +1312,217 @@ ContainsReferencesToOuterQueryWalker(Node *node, VarLevelsUpWalkerContext *conte
 }
 
 
+/*
+ * WrapFunctionsInSubqueries iterates over all the immediate Range Table Entries
+ * of a query and wraps the functions inside (SELECT * FROM fnc() f) subqueries.
+ *
+ * We currently wrap only those functions that return a single value. If a
+ * function returns records or a table we leave it as it is
```

Comment:
comment is stale

Reply:
I think it might also be nice to explain why we want to wrap things in subqueries.

Reply:
I have rewritten my stale comments, and added explanations as to why we need to wrap function calls in subqueries